src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,34 +1,73 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
+import React from 'react';
 
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
+interface Props {
+  query: string;
+  status: string;
+  onFilter: (value: string) => void;
+  onSelectStatus: (value: string) => void;
+}
 
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button
-          data-cy="clearSearchButton"
-          type="button"
-          className="delete"
+export const TodoFilter: React.FC<Props> = React.memo(({
+  query,
+  status,
+  onFilter,
+  onSelectStatus,
+}) => {
+  const handleQueryOnChange = (
+    event: React.ChangeEvent<HTMLInputElement>,
+  ) => {
+    onFilter(event.target.value);
+  };
+
+  const handleStatusSelectOnChange = (
+    event: React.ChangeEvent<HTMLSelectElement>,
+  ) => {
+    onSelectStatus(event.target.value);
+  };
+
+  const handleClearButtonOnClick = () => onFilter('');
+
+  return (
+    <form className="field has-addons">
+      <p className="control">
+        <span className="select">
+          <select
+            data-cy="statusSelect"
+            value={status}
+            onChange={handleStatusSelectOnChange}
+          >
+            <option value="all">All</option>
+            <option value="active">Active</option>
+            <option value="completed">Completed</option>
```

Comment:
Can we create a helper const or enum with our statuses and then just map it here? In the future, if we decide to add one more status, it will be easier.